MonoGame.Framework/Vector3.cs

```diff
@@ -522,7 +530,7 @@ public bool Equals(Vector3 other)
         /// <returns>Hash code of this <see cref="Vector3"/>.</returns>
         public override int GetHashCode()
         {
-            return (int)(this.X + this.Y + this.Z);
+            return X.GetHashCode() + Y.GetHashCode() + Z.GetHashCode();
```

Comment:
the float generates different values than integer - for example hash code for 1.0f is 1065353216 whereas 1 integer generates 1 hash code.